tests/core/test_processor.py

```diff
@@ -241,15 +254,198 @@
+        tracker.update(UserUttered("test"))
+        tracker.update(ActionExecuted("action_reminder_reminder"))
+        tracker.update(reminder)
+
+    default_processor.tracker_store.save(tracker)
+    await default_processor._schedule_reminders(
+        tracker.events, tracker, default_channel, default_processor.nlg
+    )
+
+    # cancel the sixth reminder
+    await cancel_reminder_and_check(
+        tracker, default_processor, ReminderCancelled("special"), 6, 5
+    )
+
+
+async def test_reminder_cancelled_by_entities(
+    default_channel: CollectingOutputChannel, default_processor: MessageProcessor
+):
+    reminders = default_reminder_list()
+    sender_id = uuid.uuid4().hex
+    tracker = default_processor.tracker_store.get_or_create_tracker(sender_id)
+    for reminder in reminders:
+        tracker.update(UserUttered("test"))
+        tracker.update(ActionExecuted("action_reminder_reminder"))
+        tracker.update(reminder)
+
+    default_processor.tracker_store.save(tracker)
+    await default_processor._schedule_reminders(
+        tracker.events, tracker, default_channel, default_processor.nlg
+    )
+
+    # cancel the fourth reminder
+    await cancel_reminder_and_check(
+        tracker,
+        default_processor,
+        ReminderCancelled(entities=[{"entity": "name", "value": "Bruce Wayne"}]),
+        6,
+        5,
+    )
+
+
+async def test_reminder_cancelled_by_intent(
+    default_channel: CollectingOutputChannel, default_processor: MessageProcessor
+):
+    reminders = default_reminder_list()
+    sender_id = uuid.uuid4().hex
+    tracker = default_processor.tracker_store.get_or_create_tracker(sender_id)
+    for reminder in reminders:
+        tracker.update(UserUttered("test"))
+        tracker.update(ActionExecuted("action_reminder_reminder"))
+        tracker.update(reminder)
+
+    default_processor.tracker_store.save(tracker)
+    await default_processor._schedule_reminders(
+        tracker.events, tracker, default_channel, default_processor.nlg
+    )
+
+    # cancel the third, fifth, and sixth reminder
+    await cancel_reminder_and_check(
+        tracker, default_processor, ReminderCancelled(intent="default"), 6, 3
+    )
+
+
+async def test_reminder_cancelled_all(
+    default_channel: CollectingOutputChannel, default_processor: MessageProcessor
+):
+    reminders = default_reminder_list()
+    sender_id = uuid.uuid4().hex
+    tracker = default_processor.tracker_store.get_or_create_tracker(sender_id)
+    for reminder in reminders:
+        tracker.update(UserUttered("test"))
+        tracker.update(ActionExecuted("action_reminder_reminder"))
+        tracker.update(reminder)
+
+    default_processor.tracker_store.save(tracker)
+    await default_processor._schedule_reminders(
+        tracker.events, tracker, default_channel, default_processor.nlg
+    )
+
+    # cancel all reminders
+    await cancel_reminder_and_check(
+        tracker, default_processor, ReminderCancelled(), 6, 0
+    )
 
```

Comment:
Do we have test which cover this case?

> ``ReminderCancelled`` can now cancel multiple reminders if no name is given.

Reply:
What do you mean? This _is_ the test that checks if a `ReminderCancelled()` cancels six reminders at once.

Reply:
🙈